### ISSUE TITLE
`block_volume_format` needlessly produces small wasteful f2fs partitions

### ISSUE THREAD

**Issue by Lanchon** [openwrt/fstools#6]
the choice of FS in `block_volume_format` should be user configurable, as that piece of code cannot know what FS is better for each particular usage. but we have what we have...

but for the case of openwrt `rootfs_data` partitions, it is arguably making the wrong choice:

- it chooses f2fs for partitions larger than 100MB. this is extremely wasteful: a 100MB f2fs partition has an overhead of about 40%! so you only get 60 usable megs out of your 100 megs, it's crazy. if auto switching is used at all, IMHO it should not cause overheads higher than 10%, so that would move the switch point to around 400 or 500 megs.

- it is well known that f2fs suffers from corruption when subjected to unclean shutdowns. it may be an acceptable risk for some battery-powered devices such as phones, especially when benchmarks are given more importance than user's data by manufacturers, but it is a bad choice for devices that are typically *always* shutdown uncleanly, like openwrt routers. we probably do not seen many corruption events only because the usual openwrt install writes so little to the disk; but then see next point...

- the performance advantages (in time and write amplification/wear) of f2fs are noticeable when the FS is being written to heavily. there is no advantage to using f2fs (and real disadvantages) if you write little to it. the probably calls into question whether f2fs should be used at all automatically.

to recap: maybe always choose ext4. but if not, the cutoff point of 100MB should be raised significantly.

BTW, in my openwrt port i have an integrated 4 GB eMMC with a huge partition on it. i would like to split that partition into volumes using btfs (@rootfs_data, @alt_rootfs_data, @extra) and mount the correct overlay at boot time. but i don't think i can do this in openwrt with scripting, or can i?

**Comment by Lanchon** [comment]
a suggestion...
https://github.com/openwrt/fstools/blob/08cd7083cac4bddf88459efa0881ee52858e7d0a/libfstools/common.c#L166-L169

before calling `use_f2fs`, or possibly within that function, could you maybe check if both `mkfs.*` are available? if one is missing, just use the other.

this would let port authors choose for example ext4 by simply not including `mkfs.f2fs` in their builds.
